src/App.tsx

```diff
@@ -2,24 +1,40 @@
+import React, { useState, useEffect } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
-
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+import { Pagination } from './components/Pagination';
 
 export const App: React.FC = () => {
+  const items = getNumbers(1, 42).map(n => `Item ${n}`);
+
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const [perPage, setPerPage] = useState<number>(5);
+
+  const startIndex = (currentPage - 1) * perPage;
+  const endIndex = Math.min(startIndex + perPage, items.length);
+  const currentItems = items.slice(startIndex, endIndex);
```

Comment:
You calculate `startIndex`, `endIndex` and `currentItems` every render, even if `currentPage` and `perPage` haven't changed. Consider using `useMemo` hook to optimize these calculations.